gapis/replay/builder/builder.go

```diff
@@ -316,6 +358,79 @@ func (b *Builder) CommitCommand() {
 	if pop > 0 {
 		b.instructions = append(b.instructions, asm.Pop{Count: pop})
 	}
+
+	// Lets inine any small resources next...
+	inlineThreshold := 128
+	for index := b.cmdStart; index < len(b.instructions); index++ {
+
+		if resource, ok := b.instructions[index].(asm.Resource); ok {
+
+			id := b.resourceIdxToID[resource.Index]
+
+			obj, err := database.Resolve(ctx, id)
+			if err != nil {
+				panic(log.Err(ctx, ErrInvalidResource, "Couldn't resolve inline resource"))
+			}
+			data, ok := obj.([]byte)
+			if !ok {
+				panic(log.Err(ctx, ErrInvalidResource, "Inline resource didn't resolve to byte slice"))
+			}
+
+			if len(data) <= inlineThreshold {
+
+				valuePatchUps := make([]asm.InlineResourceValuePatchUp, 0)
+				pointerPatchUps := make([]asm.InlineResourcePointerPatchUp, 0)
+
+				for donePatchups := false; donePatchups == false; {
+					if index+2 < len(b.instructions) {
+						push, pushOk := b.instructions[index+1].(asm.Push)
+						load, loadOk := b.instructions[index+1].(asm.Load)
+						store, storeOk := b.instructions[index+2].(asm.Store)
+						if storeOk {
+							if pushOk {
+								resolver := getRegionCalculatingVolatileMemoryLayout(*b)
+								ty, _, onStack := push.Value.Get(resolver)
+								// Because the number of valuePatchUps is stored in the instruction bits, we need to stay before 64 of them here.
+								if !onStack && ty == protocol.Type_VolatilePointer && len(valuePatchUps) < 64 {
+									valuePatchUps = append(valuePatchUps, asm.InlineResourceValuePatchUp{Destination: store.Destination, Value: push.Value})
+									index += 2
+									continue
+								}
+							}
+
+							if loadOk {
+								resolver := getRegionCalculatingVolatileMemoryLayout(*b)
+								ty1, _, onStack := load.Source.Get(resolver)
+								if !onStack {
+									ty2, _, onStack := store.Destination.Get(resolver)
+									if !onStack && ty1 == protocol.Type_VolatilePointer && ty2 == protocol.Type_VolatilePointer {
+										pointerPatchUps = append(pointerPatchUps, asm.InlineResourcePointerPatchUp{Destination: store.Destination, Source: load.Source})
+										index += 2
+										continue
+									}
+								}
+							}
+						}
+					}
+
+					donePatchups = true
+				}
+
+				inlineResource := asm.InlineResource{Ctx: ctx, Data: data, Destination: resource.Destination, ValuePatchUps: valuePatchUps, PointerPatchUps: pointerPatchUps}
+				newInstructions = append(newInstructions, inlineResource)
+			} else {
+				newInstructions = append(newInstructions, b.instructions[index])
+			}
+		} else {
+			newInstructions = append(newInstructions, b.instructions[index])
+		}
+	}
+
+	b.instructions = b.instructions[:b.cmdStart]
+	for _, newInstruction := range newInstructions {
+		b.instructions = append(b.instructions, newInstruction)
```

Comment:
b.instructions = append(b.instructions, newInstructions...)

Reply:
That's not the same? We want to drop all the "b.instructions" beyond b.cmdStart

Reply:
done